scripts/generate-renode-scripts.py

```diff
@@ -0,0 +1,168 @@
+    base = 0x20228000
+    return [
+        'boot_flash: Memory.ArrayMemory @ sysbus {}'.format(base),
+        '    size: 0x00100000'
+        ]
+
+def generate_cpu(data):
+    kind = get_entry_value(data, 'constant', 'config_cpu_type')
+
+    if kind == 'VEXRISCV':
+        return ['cpu: CPU.VexRiscv @ sysbus']
+    elif kind == 'PICORV32':
+        return [
+            'cpu: CPU.PicoRV32 @ sysbus',
+            '    cpuType: "rv32imc"'
+        ]
+    else:
+        raise Exception('Unsupported cpu type: {}'.format(kind))
+
+def get_entry_value(data, kind, name, idx=0):
+    for entry in data:
+        if entry[0] == kind and entry[1] == name:
+            return entry[idx + 2]
+    return None
+
+def generate_repl(data):
+    return \
+        generate_cpu(data) + \
+        generate_memory_region(data, 'rom') + \
+        generate_memory_region(data, 'sram') + \
+        generate_memory_region(data, 'main_ram') + \
+        generate_flash_boot(data) + \
+        generate_peripheral(data, 'uart', 'UART.LiteX_UART') + \
+        generate_peripheral(data, 'timer0', 'Timers.LiteX_Timer',
+                           {'frequency': get_entry_value(data, 'constant', 'system_clock_frequency')}) + \
+        generate_ethmac(data) + \
+        generate_silencer(data, 'ddrphy') + \
+        generate_silencer(data, 'sdram') + \
+        generate_silencer(data, 'ethphy')
+
+def generate_resc(data, host_tap_interface=None, firmware_binary=None, bios_binary=None, repl_file=None):
+    cpu_type = get_entry_value(data, 'constant', 'config_cpu_type')
+
+    result = [
+        'using sysbus',
+        'mach create "litex-{}"'.format(cpu_type),
+        'machine LoadPlatformDescription @{}'.format(repl_file),
+        'cpu StartGdbServer 10001',
+        'showAnalyzer sysbus.uart',
+        'showAnalyzer sysbus.uart Antmicro.Renode.Analyzers.LoggingUartAnalyzer',
+    ]
+
+    rom_base = get_entry_value(data, 'memory_region', 'rom')
+    if rom_base is not None and bios_binary:
+        result.extend([
+            'sysbus LoadBinary @{} {}'.format(bios_binary, rom_base),
+            'cpu PC {}'.format(rom_base)
+        ])
+
+    if host_tap_interface:
+        result.extend([
+            'emulation CreateSwitch "switch"',
+            'emulation CreateTap "{}" "tap"'.format(host_tap_interface),
+            'connector Connect ethmac switch',
+            'connector Connect host.tap switch'
+        ])
+    elif firmware_binary:
+        # todo this is hardcoded!
+        flash_boot_address = 0x20228000
+
+        firmware_data = open(firmware_binary, 'rb').read()
+        crc32 = zlib.crc32(firmware_data)
+
+        result.extend([
+            'sysbus WriteDoubleWord {} {}'.format(hex(flash_boot_address), hex(len(firmware_data))),
+            'sysbus WriteDoubleWord {} {}'.format(hex(flash_boot_address + 4), hex(crc32)),
+            'sysbus LoadBinary @{} {}'.format(firmware_binary, hex(flash_boot_address + 8))
+        ])
```

Comment:
Can you add a comment about what this is doing and why?

Reply:
This fragment of code is temporarily removed. See comments above.